### ISSUE TITLE
Apps Suddenly Quit Working on iOS 9

### ISSUE THREAD

**Issue by Bilbosax** [airsdk/Adobe-Runtime-Support#1228]
I have been using AIR 33.1.1.406 for awhile now and have not had any problems.  I have 2 iPads that are running iOS 9.3.6.  Absolutely nothing has changed.  I have not installed a new version of my apps on either device.  But my AIR apps have suddenly quit running.

This is a real mystery.  I have not added a new ANE or tried a new AIR SDK.  The versions that are in the APP STORE that I developed using one of these iPads used to work on my machines, and they suddenly do not.  This is strange because Apple doesn't update these iPads anymore, so iOS should not have changed.  So what could have happened?

I thought perhaps it was my backend server, so I recompiled the apps to debug them, and no errors are thrown, but they are not able to pull data from my servers using AMF anymore.  I know that nothing is wrong with the servers because they still work with newer iPads, phones, and desktop apps.  So it has to have something to do with these iPads.

I might have thought that I had done something wrong if it was just on one iPad, but it is happening on two with the same operating system.

Any thoughts???

**Comment by FliplineStudios** [comment]
Any chance your server is using LetsEncrypt for its SSL certificate?  Last week there's was a big change with LetsEncrypt certificates including an expiring intermediate certificate.  Our server was using LetsEncrypt and it caused our AIR apps to stop communicating with the server on any devices running iOS 9.3.6 or earlier, while later iOS versions continued to work fine (though it also caused full-screen popup errors on all versions of Android saying the server could not be verified...).  In the end we just got a separate SSL certificate to use instead of relying on LetsEncrypt any longer, since there didn't seem to be any great workarounds for apps built using OpenSSL 1.0.2 (basically all of our AIR apps).

**Comment by Bilbosax** [comment]
@FliplineStudios I absolutely used LetsEncrypt.  If there is no known workaround, can you please tell me where you got your certificate that was compatible with older versions of AIR and iOS??

Thanks

**Comment by FliplineStudios** [comment]
Our hosting company ended up configuring SSL through CloudFlare since we were already using it for other services, but we were planning to use one from rapidSSL before we realized the CloudFlare setup would quickly solve our problem.  Before migrating servers we used to have a GoDaddy SSL certificate on that server too, which had also worked fine with our apps.

**Comment by FliplineStudios** [comment]
If you also have Android versions of your apps, I'd give them a check as well to see if you get a full-screen scary warning popup about an unverified server and requiring you to click a "Trust for Session" button -- if you're seeing that warning too (and see Issued By: R3, Org: Let's Encrypt in that popup window), then that's likely the problem!

**Comment by Bilbosax** [comment]
@FliplineStudios Yep, big scary warning on Android devices ;)  Must be the certificate.  I wish I was more confident about certificates. I know nothing about cloudflare, I am sure that it is a fine solution, but I think I would rather pay and have peace of mind for a year.  My other free option LetsEncrypt didn't work out so well LOL  I'm thinking Sectigo or Godaddy

**Comment by Michoko92** [comment]
I'm having the same issue with all my apps on Android (and Amazon which uses Android too)! I really don' t want to republish all my apps 🙁

I'm not very knowledgeable about SSL certificates, but my server definitely uses a Let's encrypt one. It's a mutualized hosting service, and I contacted the support team to ask them to regenerate it, but unfortunately it didn't solve the problem. Does that mean that any Let's encrypt certificate will have this issue with AIR apps? 

I doubt my host will accept to get a certificate from another entity if they usually get it from Let's encrypt. Do you have any suggestion or at least further info about this Let's encrypt global issue I could forward to my provider? Thank you so much for your help!

**Comment by FliplineStudios** [comment]
@Michoko92 We were in the same boat, the hosting company regenerated the certificate for us but it didn't make any difference. I did find this information that OpenSSL published about the LetsEncrypt issue with anything using OpenSSL 1.0.2 like AIR:
https://www.openssl.org/blog/blog/2021/09/13/LetsEncryptRootCertExpire/

It sounds like "Workaround 3" would be the only feasible change that can be made since it's a server-side change rather than an app-based change, though I'm not sure how successful it would be, and they do also note a downside to that method.  I don't believe our host tried to go down that path, since we were basically in damage control mode and were looking for a quick fix to stop all the emails coming in asking about the error popup :)  

The LetsEncrypt forum has more information of others trying to troubleshoot (and it was pretty awful that LetsEncrypt's "support" for this was just telling people to check the forum to see what others figure out):
https://community.letsencrypt.org/t/help-thread-for-dst-root-ca-x3-expiration-september-2021/149190
I'd recommend starting around post 300 on Sept 30, though I'm not sure how helpful any of that troubleshooting might be.

**Comment by ajwfrost** [comment]
Not sure if it will help but AIR updated to openssl 1.1.1 around March/April time ... AIR 33.1.1.476 (or later) should be used to get this.

**Comment by Michoko92** [comment]
@FliplineStudios thank you for your help (and your constant contribution to the community, you are always very helpful!). I contacted my host, but honestly, I don't have high hopes about what they could do. I think my only solution would be to re-publish more than 30 apps on both the Play Store and Amazon Store, which is a huge pain... I really pray that the issue won't appear on recent iOS devices too.

@ajwfrost Thank you, I confirm that the issue seems to disappear when I compile my apps with a recent AIR SDK. Thank you for providing a solution with your great updates to the AIR SDK!
